hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsContext.java

```diff
@@ -142,7 +142,7 @@ public void notifyListeners(Set<DefinitionsEvent> notifications) {
                     log.debugf("Notified Listener %s of %s", e.getKey(), notificationTypes);
                     e.getKey().onChange(notifications.stream()
                             .filter(de -> e.getValue().contains(de.getType()))
-                            .collect(Collectors.toSet()));
+                            .collect(Collectors.toCollection(LinkedHashSet::new)));
```

Comment:
I guess this could fix it but isn't it also dangerous that notifyListeners itself takes a Set?  If we expect something ordered then maybe we should be using Lists, or maybe notifyListeners should specifically take a LinkedHashSet.